docs/4.4/api-reference.md

```diff
@@ -11,57 +11,287 @@ Teleport is currently working on documenting our API.
+
+```
+.
++-- api-admin.yaml
++-- certs
+|   +-- api-admin.cas
+|   +-- api-admin.crt
+|   +-- api-admin.key
++-- client.go
++-- main.go
+```
+
+**client.go**
+```go
+{!examples/go-client/client.go!}
+```
+
+## Roles
+
+[Roles](enterprise/ssh-rbac.md#roles) are used to define what resources and actions a user is allowed/denied access to. It's best to make your role's permissions as stringent as possible.
+
+**Create a new role**
+
+```go
+// create a new auditor role which has very limited permissions
+role, err := services.NewRole("auditor", services.RoleSpecV3{
+  Options: services.RoleOptions{
+    MaxSessionTTL: services.Duration(time.Hour),
+  },
+  Allow: services.RoleConditions{
+    Logins: []string{"auditor"},
+    Rules: []services.Rule{
+      // - resources: ['session']
+      //   verbs: ['list', 'read']
+      services.NewRule(services.KindSession, services.RO()),
+    },
+  },
+  Deny: services.RoleConditions{
+    // node_labels: '*': '*'
+    NodeLabels: services.Labels{"*": []string{"*"}},
+  },
+})
+if err != nil {
+  return err
+}
+
+if err := client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+**Retrieve role**
+
+```go
+role, err := client.GetRole("auditor")
+if err != nil {
+  return err
+}
+```
+
+**Update role**
+
+```go
+// update the auditor role's read rule to not provide access to secrets
+role.SetRules(services.Allow, []services.Rule{
+	services.NewRule(services.KindSession, services.ReadNoSecrets()),
+})
+if err := client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+**Delete role**
+
+```go
+if err := client.DeleteRole(ctx, "auditor"); err != nil {
+  return err
+}
+```
+
+## Tokens
+
+[Tokens](admin-guide.md#adding-nodes-to-the-cluster) are used to add nodes to clusters, and add remote clusters to [trusted clusters](admin-guide.md#trusted-clusters).
+
+**Create token**
+
+```go
+tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
+  // You can provide 'Token' for a non-random tokenString
+  Roles: teleport.Roles{teleport.RoleProxy},
+  TTL:   time.Hour,
+})
+if err != nil {
+  return err
+}
+```
+
+**Retrieve token**
 
-## Audit Events API
-Coming Soon
+```go
+token, err := client.GetToken(tokenString)
+if err != nil {
+  return err
+}
+```
+
+**Update token**
+
+```go
+// updates the token to be a proxy token
+token.SetRoles(teleport.Roles{teleport.RoleProxy})
+if err := client.UpsertToken(token); err != nil {
+  return err
+}
+```
+
+**Delete token**
+
+```go
+if err := client.DeleteToken(tokenString); err != nil {
+  return err
+}
+```
+
+## Cluster Labels
+
+The root cluster in a [trusted cluster](admin-guide.md#adding-nodes-to-the-cluster) can add/update cluster labels on remote clusters.
+
+**Create a remote cluster token with Labels**
```

Comment:
We should add a note, that this is only supported once https://github.com/gravitational/teleport/pull/4630 is merged. ( I'm not sure if we'll backport to 4.4 )

Reply:
Is there a specific release we can mention, like 5.0 Beta or something?

Reply:
Correct, it'll only be 5.0. I would just leave a note in the docs for now.